build-logic/src/main/groovy/io.vavr.library.gradle

```diff
@@ -0,0 +1,43 @@
+/**
+ * This plugin configures the vavr library: compilation options, jacoco support,
+ * additional artifacts like test sources jar, automatic module name...
+ */
+plugins {
+    id 'java-library'
+    id 'jacoco'
+}
+
+java {
+    toolchain {
+        languageVersion.set(JavaLanguageVersion.of(8))
```

Comment:
@melix One question: it is common practice to compile & test a project with different JDKs (-> travis build matrix) in order to see if the code behaves as expected with different Java versions.

Currently, each build of our Travis CI matrix falls back to the alternate Java 8 JDK:

```bash
Compiling with toolchain '/usr/lib/jvm/java-8-openjdk-amd64'
```

Do you have any suggestion what's the best way to circumvent this?

I have in mind: using an optional env variable that defaults to `8`.

Reply:
I have this in mind:

```
def val = findProperty('java-language-version') ?: '8'
```

Reply:
Ah, I've found the Gradle toolchain docs --> https://docs.gradle.org/current/userguide/toolchains.html

Reply:
There's no built-in support in Gradle for test matrix but it's a common practice. It's usually a waste of time and resources, though, to rebuild the project with a different JDK just to test on a different one. So the simplest solution is what you've described, which can be implemented as this:

```
    toolchain {
        languageVersion.set(JavaLanguageVersion.of(
            providers.environmentVariable("JAVA_VERSION")
                .forUseAtConfigurationTime()
                .map { it.toInt() } // Kotlin!!
                .getOrElse(8)
        ))
    }
```

Alternatively, you can configure different test tasks to use different JDKs, which is what I would personally prefer, e.g (sorry it's Kotlin DSL because I just copied it from a project of mine):

```
tasks.register<Test>("java11Test") {
    group = "verification"
    val test = tasks.test.get()
    classpath = test.classpath
    testClassesDirs = test.testClassesDirs
    javaLauncher.set(javaToolchains.launcherFor {
        languageVersion.set(JavaLanguageVersion.of(11))
    })
}
```

Reply:
Nice one, thanks!